types/index.d.ts

```diff
@@ -183,6 +187,138 @@ declare module '@azure/functions' {
      * Possible values for an HTTP Request user type
      */
     export type HttpRequestUserType = 'AppService' | 'StaticWebApps';
+    /**
+     * HTTP response object
+     */
+    export interface IResponse {
```

Comment:
Wow, that was an interesting rabbit hole to go down lol, thanks for sharing. I just named it `IResponse` just because that's what `Context.res` used before. I didn't know it was that deep lol. Anyway I changed it to `HttpResponseBase` now.